src/common/components/repositories-home/index.js

```diff
@@ -55,7 +55,10 @@ class RepositoriesHome extends Component {
   }
 
   componentWillReceiveProps(nextProps) {
-    if (this.props.snaps.success !== nextProps.snaps.success) {
+    const snapsLength = nextProps.snaps.snaps && nextProps.snaps.snaps.length;
+
+    if ((this.props.snaps.success !== nextProps.snaps.success)
+        || (snapsLength === 0)) {
```

Comment:
Make sure user is redirected to select repositories view when all snaps are removed.